apps/server/src/shared/infra/identity-management/keycloak-configuration/service/keycloak-configuration.service.integration.spec.ts

```diff
@@ -0,0 +1,170 @@
+import { createMock } from '@golevelup/ts-jest';
+import KeycloakAdminClient from '@keycloak/keycloak-admin-client';
+import AuthenticationExecutionExportRepresentation from '@keycloak/keycloak-admin-client/lib/defs/authenticationExecutionExportRepresentation';
+import AuthenticationFlowRepresentation from '@keycloak/keycloak-admin-client/lib/defs/authenticationFlowRepresentation';
+import { ConfigModule } from '@nestjs/config';
+import { Test, TestingModule } from '@nestjs/testing';
+import { MongoMemoryDatabaseModule } from '@shared/infra/database';
+import { SystemRepo } from '@shared/repo';
+import { systemFactory } from '@shared/testing/factory';
+import { LoggerModule } from '@src/core/logger';
+import { SystemMapper } from '@src/modules/system/mapper/system.mapper';
+import { SystemService } from '@src/modules/system/service/system.service';
+import { KeycloakAdministrationService } from '../../keycloak-administration/service/keycloak-administration.service';
+import { KeycloakConfigurationModule } from '../keycloak-configuration.module';
+import { KeycloakConfigurationService } from './keycloak-configuration.service';
+
+describe('KeycloakConfigurationService Integration', () => {
+	let module: TestingModule;
+	let keycloak: KeycloakAdminClient;
+	let keycloakAdministrationService: KeycloakAdministrationService;
+	let keycloakConfigurationService: KeycloakConfigurationService;
+	let isKeycloakAvailable = false;
+
+	const testRealm = 'test-realm';
+	const flowAlias = 'Direct Broker Flow';
+	const systemServiceMock = createMock<SystemService>();
+	const systems = SystemMapper.mapFromEntitiesToDtos(systemFactory.withOidcConfig().buildList(1));
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			imports: [
+				KeycloakConfigurationModule,
+				LoggerModule,
+				MongoMemoryDatabaseModule.forRoot(),
+				ConfigModule.forRoot({
+					isGlobal: true,
+					validationOptions: { infer: true },
+				}),
+			],
+			providers: [SystemRepo],
```

Comment:
Since `SystemService` is mocked the `SystemRepo` is not needed here, right?

Reply:
done